api/docker.ml

```diff
@@ -94,14 +95,12 @@ module Spec = struct
     let include_git = R.include_git_get r in
     let options = { build_args; squash; buildkit; include_git } in
     let push_to =
-      match target, user, password with
-      | "", "", "" -> None
-      | "", _, _
-      | _, "", _
-      | _, _, "" -> Fmt.failwith "push-target, push-user and push-password must be given (or not given) together"
-      | target, user, password ->
+      if user = "" && password <> "" || user <> "" && password = "" then
+        Fmt.failwith "push-user and push-password must be given (or not given) together"
+      else
```

Comment:
Looks like this is missing a check for auth but no target.

Reply:
Eek - and also the no push-target, push-user, push-password case!